src/librustc_interface/passes.rs

```diff
@@ -719,31 +720,49 @@ pub fn prepare_outputs(
     Ok(outputs)
 }
 
-pub fn default_provide(providers: &mut ty::query::Providers) {
-    providers.analysis = analysis;
-    proc_macro_decls::provide(providers);
-    plugin::build::provide(providers);
-    rustc_middle::hir::provide(providers);
-    mir::provide(providers);
-    mir_build::provide(providers);
-    rustc_privacy::provide(providers);
-    typeck::provide(providers);
-    ty::provide(providers);
-    traits::provide(providers);
-    rustc_passes::provide(providers);
-    rustc_resolve::provide(providers);
-    rustc_traits::provide(providers);
-    rustc_ty::provide(providers);
-    rustc_metadata::provide(providers);
-    rustc_lint::provide(providers);
-    rustc_symbol_mangling::provide(providers);
-    rustc_codegen_ssa::provide(providers);
-}
+pub const DEFAULT_QUERY_PROVIDERS: Providers = {
+    // FIXME(eddyb) the `const fn` shouldn't be needed, but mutable references
```

Comment:
This rads as if it should be a `fn` instead, but I think what you actually mean is that it should be directly inlined in `DEFAULT_QUERY_PROVIDERS` instead?

Reply:
Indeed, maybe I should say "separate function" or something like that.